src/rdkafka_cgrp.c

```diff
@@ -5658,20 +5720,7 @@ void rd_kafka_cgrp_consumer_group_heartbeat(rd_kafka_cgrp_t *rkcg,
                 member_epoch = 0;
 
 
-        if (rkcg->rkcg_flags & RD_KAFKA_CGRP_F_MAX_POLL_EXCEEDED) {
-                if (rd_kafka_max_poll_exceeded(rkcg->rkcg_rk)) {
-                        /* Don't send heartbeats if max.poll.interval.ms was
-                         * exceeded */
-                        return;
-                } else {
-                        rkcg->rkcg_flags &= ~RD_KAFKA_CGRP_F_MAX_POLL_EXCEEDED;
-                }
-        }
-
-        /* Skip heartbeat if we have one in transit */
-        if (rkcg->rkcg_flags & RD_KAFKA_CGRP_F_HEARTBEAT_IN_TRANSIT)
```

Comment:
Why removing this?

Reply:
Moved to `rd_kafka_cgrp_consumer_heartbeat_preconditions_met`